### PR TITLE
static support mp_layers

### PR DESCRIPTION
**Pull request by wangxicoding**
### PR types

New features
### PR changes

APIs
### Describe

c_softmax_with_cross_entropy API support static graph.
mp_layers接口支持静态图，mp_layers实现动静统一。
``` python
import paddle
import paddle.distributed.fleet as fleet

paddle.enable_static()

class ColumnLinearNet(paddle.nn.Layer):
    def __init__(self, input_size, output_size):
        super(ColumnLinearNet, self).__init__()
        self.parallel_linear = fleet.meta_parallel.ColumnParallelLinear(
            in_features=input_size,
            out_features=output_size,
            weight_attr=None,
            has_bias=True,
            gather_output=True,
            name="test_column_linear")

    def forward(self, x):
        output = self.parallel_linear(x)
        return output

strategy = fleet.DistributedStrategy()
strategy.sharding = True
strategy.sharding_configs = {
    "mp_degree": 2,
    "sharding_degree": 2,
}
fleet.init(is_collective=True, strategy=strategy)

input_size, output_size = 28, 64
model_a = ColumnLinearNet(input_size, output_size)

x = paddle.static.data(name='x', shape=[None, input_size])
y = model_a(x)
```
`python -m paddle.distributed.launch --gpus 0,1,2,3 test.py`

### CODE REVIEW

**Review by kuizhiqing — APPROVED** [review]
lgtm for new group part

File: python/paddle/distributed/fleet/base/fleet_base.py [hunk 1]
```diff
@@ -253,6 +253,40 @@ def init(self, role_maker=None, is_collective=False, strategy=None):
                 warnings.warn(
                     "The dygraph hybrid parallel environment has been initialized."
                 )
+        elif self._is_collective:
+            use_sharding = self._user_defined_strategy.sharding
+
+            # global group
+            global_rank = self.worker_index()
+            global_world_size = self.worker_num()
+            # NOTE(wangxi): see sharding_optimizer
```
Comment (JZ-LIANG) [inline python/paddle/distributed/fleet/base/fleet_base.py:262]: or we could explain the parallel2id mapping here again?

**Review by JZ-LIANG — APPROVED** [review]
LGTM

File: python/paddle/distributed/fleet/base/fleet_base.py (hunk 1 above)
Comment (JZ-LIANG) [reply python/paddle/distributed/fleet/base/fleet_base.py:262]: pre-assign ring ids
            mp: 0
            sharding: 1
            pure-dp: 2
            global: 3
            pp: >= 20
        if one parallelism is not enable: -1
        and only support parallelism hierarchy: mp --> sharding --> pp --> dp      


done

**Review by ForFishes — APPROVED** [review]
LGTM

**Pull request merged**